### PR TITLE
Add SDLAda v2.5.4 Alire TOML.

### PR DESCRIPTION
**Pull request by Lucretia**
Here, is an official Alire TOML file for SDLAda.

### CODE REVIEW

**Comment by Lucretia** [comment]
2.5.5 is in the works, but I'm getting:

```
Publishing assistant: step 4 of 6: Build release
error: Cannot continue with invalid session:
error:    Loading release from manifest: alire.toml:
error:    Failed to load alire.toml:
error:    Identifiers must be lowercase ASCII alphanumerical. You can see the complete identifier naming rules with 'alr help identifiers'
error: Could not complete the publishing assistant:
error:    Command ["alr", "--non-interactive", "build"] exited with code 1
```

Which doesn't tell me where it's going wrong.

**Comment by Lucretia** [comment]
Debugging:

```
$ alr -d -v publish . v2.5.5 
-> Config file is not readable or doesn't exist: '/home/laguest/src/mine-new/game-dev/sdlada/alire/config.toml'
-> Using default value for configuration 'distribution.disable_detection': 'FALSE'
-> Spawning: ["cat", "/etc/os-release"] > GNAT-TEMP-000001.TMP
detail: Spawning: ["git", "status"] > GNAT-TEMP-000002.TMP
detail: Spawning: ["git", "status", "--porcelain"] > GNAT-TEMP-000003.TMP
detail: Spawning: ["git", "remote"] > GNAT-TEMP-000004.TMP
detail: Spawning: ["git", "branch"] > GNAT-TEMP-000005.TMP
detail: Spawning: ["git", "rev-list", "origin/master..HEAD"] > GNAT-TEMP-000006.TMP
info: ✓ Local repository is clean.
detail: Spawning: ["git", "log", "v2.5.5", "-n1", "--oneline", "--no-abbrev-commit"] > GNAT-TEMP-000007.TMP
info: ✓ Revision exists in local repository (429c594de613c5ba2f0d7c59f8708956697e14f1).
detail: Spawning: ["git", "remote"] > GNAT-TEMP-000008.TMP
detail: Spawning: ["git", "config", "--list"] > GNAT-TEMP-000009.TMP
warn: Private git git@github.com:Lucretia/sdlada.git transformed to public https://github.com/Lucretia/sdlada.git

info: Publishing assistant: step 1 of 6: Verify origin URL
info: ✓ Origin is of supported kind: GIT
info: ✓ Origin is hosted on trusted site: github.com

info: Publishing assistant: step 2 of 6: Verify GitHub infrastructure
info: ✓ User has a GitHub account: Lucretia
info: ✓ User has forked the community repository
info: ✓ User's fork contains base branch: stable-1.1

info: Publishing assistant: step 3 of 6: Deploy sources
detail: Checking out [git]: https://github.com/Lucretia/sdlada.git#429c594de613c5ba2f0d7c59f8708956697e14f1
detail: Spawning: ["git", "clone", "--recursive", "--progress", "https://github.com/Lucretia/sdlada.git", "/home/laguest/src/mine-new/game-dev/sdlada/alire/tmp/alr-seaw.tmp"]
Cloning into '/home/laguest/src/mine-new/game-dev/sdlada/alire/tmp/alr-seaw.tmp'...
remote: Enumerating objects: 2882, done.
remote: Counting objects: 100% (52/52), done.
remote: Compressing objects: 100% (35/35), done.
remote: Total 2882 (delta 23), reused 35 (delta 14), pack-reused 2830
Receiving objects: 100% (2882/2882), 634.09 KiB | 4.80 MiB/s, done.
Resolving deltas: 100% (2126/2126), done.
detail: Spawning: ["git", "checkout", "-q", "429c594de613c5ba2f0d7c59f8708956697e14f1"]
detail: Spawning: ["git", "submodule", "update", "--init", "--recursive", "--progress"]

info: Publishing assistant: step 4 of 6: Build release
detail: Spawning: ["alr", "--non-interactive", "build"]
ⓘ Synchronizing workspace...
Nothing to update.                                                       
error: failed precondition from alire-conditional_trees.ads:97 instantiated at alire-conditional.ads:18
error: alr encountered an unexpected error, re-run with -d for details.
stderr: Command ["alr", "--non-interactive", "build"] exited with code 1
stderr: ALIRE.CHECKED_ERROR
stderr: Command ["alr", "--non-interactive", "build"] exited with code 1
stderr: raised ALIRE.CHECKED_ERROR : alire-stored-error:1
[/home/laguest/opt/bin/alr]
Alire.Raise_Checked_Error at alire.adb:232
Alire.Os_Lib.Subprocess.Checked_Spawn at alire-os_lib-subprocess.adb:110
Alire.Publish.Check_Build at alire-publish.adb:274
Alire.Publish.Run_Steps at alire-publish.adb:853
Alire.Publish.Remote_Origin at alire-publish.adb:1050
Alire.Publish.Local_Repository at alire-publish.adb:992
Alr.Commands.Publish.Execute at alr-commands-publish.adb:80
Alr.Commands.Sub_Cmd.Execute at clic-subcommand-instance.adb:685
Alr.Commands.Execute at alr-commands.adb:410
Alr.Main at alr-main.adb:27
Main at b__alr-main.adb:2014
[/lib64/libc.so.6]
0x7fda9bb71318
0x7fda9bb713c6
[/home/laguest/opt/bin/alr]
0x42c3c8 _start at ???
0xfffffffffffffffe

stderr: Could not complete the publishing assistant
Command ["alr", "--non-interactive", "build"] exited with code 1
error: Could not complete the publishing assistant:
error:    Command ["alr", "--non-interactive", "build"] exited with code 1
```

**Comment by Lucretia** [comment]
I tried to submit a newer toml for 2.5.5 (see above) before the deadline and would've got it in if it wasn't for the errors above, I've no idea what is going on there. This one was tested on GCC 11 under Fedora and Macos Homebrew.

File: index/sd/sdlada/sdlada-2.5.4.toml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+# project-files = ["build/gnat/tools.gpr", "build/gnat/sdlada.gpr"] #, "build/gnat/"]
+project-files = ["build/gnat/sdlada.gpr"] #, "build/gnat/"]
+
+[gpr-externals]
+SDL_PLATFORM = ["linux", "bsd", "windows", "macosx", "ios", "android"]
+SDL_MODE = ["debug", "release"]
+
+[gpr-set-externals]
+BUILD_MODE = "release"
```
Comment (mosteo) [inline index/sd/sdlada/sdlada-2.5.4.toml:20]: We recommend to not force a build mode so dependents can choose the one they prefer, and your project file may use one by default if none given.

Is this related to the `SDL_MODE` in line 17?

**Comment by mosteo** [comment]
In regard to this error:

> error: failed precondition from alire-conditional_trees.ads:97 instantiated at alire-conditional.ads:18

This is indeed a bug in Alire, but the trigger I suspect is an unintended definition in your manifest. The lines 32-33
```
[depends-on.'case(os)'.linux]
    [depends-on.'case(distribution)'.'debian|ubuntu']
```
seem to imply a nested table, but in reality is creating an empty `depends-on.case(os).linux` table which is the one causing the problem. (The `distribution` case is not nested under `os`.) Guessing that you want to keep the `os` case for when Windows/Mac are readded, you need to give the whole path in the second nested table: 
```
[depends-on.'case(os)'.linux]
    [depends-on.'case(os).linux.'case(distribution)'.'debian|ubuntu']
```
I've tested and with this change the publishing assistant can proceed.

**Comment by Fabien-Chouteau** [comment]
@Lucretia Why don't you start from the existing manifest:
```TOML
name = "sdlada"
version = "2.5.4-1"
description = "Ada 2012 bindings to SDL 2"
licenses = "Zlib"
tags = ["sdl", "framework"]
website = "https://github.com/Lucretia/sdlada"

authors = ["Luke A. Guest"]
maintainers = ["alejandro@mosteo.com"]
maintainers-logins = ["mosteo"]

project-files = ["build/gnat/sdlada.gpr"]

[[depends-on]]
libsdl2 = "^2.0"
libsdl2_image = "^2.0"
libsdl2_ttf = "^2.0"
# gnat = "/=2020" # Does not build yet with CE 2020

[[actions."case(os)".linux]]
type = "post-fetch"
command = ["make", "-C", "build/gnat", "SDL_PLATFORM=linux"]

[[actions."case(os)".linux]]
type = "pre-build"
command = ["make", "-C", "build/gnat", "SDL_PLATFORM=linux"]

[[actions."case(os)".macos]]
type = "post-fetch"
command = ["make", "-C", "build/gnat", "SDL_PLATFORM=macosx"]

[[actions."case(os)".windows]]
type = "post-fetch"
command = ["make", "-C", "build/gnat", "SDL_PLATFORM=windows"]

[available."case(os)"]
'linux|macos|windows' = true
"..." = false

[gpr-externals]
SDL_MODE = ["debug", "release"]
SDL_PLATFORM = ["linux", "bsd", "windows", "macosx", "ios", "android"]

[gpr-set-externals."case(os)".linux]
SDL_PLATFORM = "linux"
[gpr-set-externals."case(os)".macos]
SDL_PLATFORM = "macosx"
[gpr-set-externals."case(os)".windows]
SDL_PLATFORM = "windows"

[environment.'case(distribution)'.msys2.C_INCLUDE_PATH]
# sdlada expects SDL2 headers to be in the C_INCLUDE_PATH but the option
# used in the project file ("-I/usr/include/SDL2") doesn't work with msys2
append = "${DISTRIB_ROOT}/mingw64/include/SDL2"

[origin]
commit = "7981bbd658fa7b0618f6fae689b3af634d2d8234"
url = "git+https://github.com/alire-project/sdlada.git"
```

**Comment by Lucretia** [comment]
I just did a publish and didn't realise it has produced a 2.5.5 not a 2.5.6 after I made the changes to the toml file.

**Comment by mosteo** [comment]
This PR should be reopened against the `stable-1.2` branch, which is the current stable. In my experience it is not enough to rebase and change the base branch, due to some limitation in the Github actions framework, so I'm closing it now. Sorry for the inconvenience.
